runtime/lua/vim/lsp.lua

```diff
@@ -542,6 +542,12 @@ function lsp.start_client(config)
       client_ids[client_id] = nil
     end
 
+    if signal ~= 0 then
```

Comment:
Isn't this mixing up signal and exit code?

Reply:
Is this better? I think it's good to inform on both signal and exit code (in case someone kills their server and wants to verify?)

Reply:
The exit code is 0 if you send sigterm to the language server

Reply:
<img width="1392" alt="image" src="https://user-images.githubusercontent.com/13316262/110019952-dcdcb300-7cdd-11eb-8f6e-0a0c08c464c9.png">

Reply:
Yes sorry, I  should've been more verbose.

I think the normal graceful exit case is signal 15 (sigterm) with exit code 0.
Currently the `or signal ~= 0` condition would lead to a notification in that scenario as well.  I don't think that's what a user wants.

Reply:
That makes sense. When you say normal graceful exit case, what are you referring to? When we send stop client, we don't use sigterm, we send a stop request to the server, so I believe in most cases  the signal should be 0 unless a user sends sigterm/kill from console

Reply:
Okay. I was confused by the `0` and didn't know that it's the value returned by luv if the process exits by itself.

Otherwise `0` is kind of a ping signal. From the [kill manual](https://man.archlinux.org/man/kill.2)

> If sig is 0, then no signal is sent, but existence and permission checks are still performed


I'd still think of `15` / `SIGTERM` as a clean intended shutdown, but no strong opposition anymore :)

Reply:
I think I agree-ish with your point, I'm inclined to merge as is because it's just a warning (I removed the language that your server crashed, changed it to quit) because it's still informative if the server didn't exit when we send the shutdown command. We can always change the behavior if people complain

Reply:
> because it's still informative if the server didn't exit when we send the shutdown command

Agreed

> I removed the language that your server crashed, changed it to quit

Nice :+1: